pkg/crypto/mock/signer.go

```diff
@@ -18,6 +18,7 @@ type signerMock struct {
 	signTx          func(transaction *types.Transaction, chainID *big.Int) (*types.Transaction, error)
 	signTypedData   func(*eip712.TypedData) ([]byte, error)
 	ethereumAddress func() (common.Address, error)
+	signature       []byte
```

Comment:
It would be convenient to have `signature` as `signFunc func([]byte) ([]byte, error)` in order to be able to validate both the input argument and error handling in tests. This change would reflect on `WithSignature` function. Also it would be inline with already existing functions in this mock.